src/d/actor/d_a_tag_so.cpp

```diff
@@ -4,66 +4,92 @@
 //
 
 #include "d/actor/d_a_tag_so.h"
+#include "d/d_lib.h"
 #include "d/d_procname.h"
 
 /* 000000EC-0000010C       .text __ct__14daTag_So_HIO_cFv */
 daTag_So_HIO_c::daTag_So_HIO_c() {
-    /* Nonmatching */
+    mNo = -1;
+    m05 = 0;
+    return;
 }
 
+static daTag_So_HIO_c l_HIO;
+
 /* 0000010C-00000114       .text _execute__10daTag_So_cFv */
 bool daTag_So_c::_execute() {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00000114-0000018C       .text debugDraw__10daTag_So_cFv */
-void daTag_So_c::debugDraw() {
-    /* Nonmatching */
+void daTag_So_c::debugDraw() {  
+    cXyz actorPos = current.pos;
+    actorPos.y += 20.0f;
+    if (m298 == 1)
+        dLib_debugDrawFan(actorPos, shape_angle.y, 0x3500, mRadius, (GXColor){0xFF, 0x00, 0x00, 0x80});
+    return;
 }
 
 /* 0000018C-000001C4       .text _draw__10daTag_So_cFv */
 bool daTag_So_c::_draw() {
-    /* Nonmatching */
+    if(l_HIO.m05)
+        debugDraw();
+    return TRUE;
 }
 
 /* 000001C4-00000224       .text getArg__10daTag_So_cFv */
 void daTag_So_c::getArg() {
-    /* Nonmatching */
+    s32 param;
+    s32 paramBit;
+    
+    param = fopAcM_GetParam(this);
+    mParam = param;
```

Comment:
```suggestion
    m290 = fopAcM_GetParamBit(param, 0, 8);
```
Because the other 2 params use this inline, we can probably assume this one does too instead of relying the implicit cast to mask it. And "mParam" is too vague of a name so it's better to leave it unnamed